client/cli/publish.py

```diff
@@ -2,90 +2,61 @@
-                    os.path.join(staging_dir, 'cli', 'common'))
-
-    # Top-level files.
-    for filename in REQUIRED_FILES:
-        filename += '.py'
-        fullname = os.path.join(OK_ROOT, filename)
-        shutil.copyfile(fullname, os.path.join(staging_dir, filename))
-
-    for folder in REQUIRED_FOLDERS:
-        shutil.copytree(os.path.join(OK_ROOT, folder),
-                        os.path.join(staging_dir, folder))
-
-    populate_protocols(staging_dir)
-    populate_sources(staging_dir)
-
-def populate_protocols(staging_dir):
-    """Populates the protocols/ directory in the staging directory with
-    relevant protocols.
-    """
-    shutil.copytree(os.path.join(OK_ROOT, 'protocols'),
-                    os.path.join(staging_dir, 'protocols'))
+def abort(message):
+    print(message + ' Aborting', file=sys.stderr)
+    sys.exit(1)
 
-def populate_sources(staging_dir):
-    """Populates the sources/ directory in the staging directory with
-    relevant sources.
-    """
-    shutil.copytree(os.path.join(OK_ROOT, 'sources'),
-                    os.path.join(staging_dir, 'sources'))
+def find_site_packages_directory():
+    virtualenv = os.getenv('VIRTUAL_ENV')
+    if not virtualenv:
+        abort('You must activate your virtualenv to publish.')
 
-def create_zip(staging_dir, destination):
-    if not os.path.isdir(destination):
-        os.mkdir(destination)
+    for path in sys.path:
+        if path.startswith(virtualenv) and 'site-packages' in path:
+            return path
 
-    dest = os.path.join(destination, client.FILE_NAME)
-    zipf = zipfile.ZipFile(dest, 'w')
-    zipf.write(os.path.join(OK_ROOT, '__main__.py'), './__main__.py')
-    for root, _, files in os.walk(staging_dir):
-        if '__pycache__' in root:
-            continue
+    abort('No site packages directory found.')
+
+def write_tree(zipf, src_directory, dst_directory):
+    """Write all .py files in a source directory to a destination directory
+    inside a zip archive.
+    """
+    for root, _, files in os.walk(src_directory):
         for filename in files:
-            if filename.endswith('.pyc'):
+            if not filename.endswith('.py'):
                 continue
             fullname = os.path.join(root, filename)
-            # Replace 'staging' with './client' in the zip archive.
-            arcname = fullname.replace(staging_dir, './client')
+            arcname = fullname.replace(src_directory, dst_directory)
             zipf.write(fullname, arcname=arcname)
-    zipf.close()
+
+def package_client(destination):
+    package_dir = find_site_packages_directory()
+
+    if not os.path.isdir(destination):
+        os.mkdir(destination)
+    dest = os.path.join(destination, client.FILE_NAME)
+
+    with zipfile.ZipFile(dest, 'w') as zipf:
+        zipf.write(os.path.join(OK_ROOT, '__main__.py'), '__main__.py')
+        write_tree(zipf, OK_ROOT, 'client')
+        for package in EXTRA_PACKAGES:
```

Comment:
It would only be an issue if we include a package called `client` in `EXTRA_PACKAGES`. The chance of that is pretty low